src/mscorlib/src/System/Environment.cs

```diff
@@ -1519,8 +1583,11 @@ public static string UserDomainName
         {
             [System.Security.SecuritySafeCritical]  // auto-generated
             get {
+#if PLATFORM_UNIX
+                return MachineName;
```

Comment:
> If there's no "redmond\danmose" I think I would expect the user name "danmose"

I think the name of the property might be throwing you off.  It's not a user name that includes a domain name; the "user" part is just in contrast to "machine", i.e. this is the domain name associated with the user's account rather than the domain name associated with the machine.  Even on Windows, this falls back to being the machine name... from the MSDN docs:
```
The UserDomainName property first attempts to get the domain name component of the Windows account name for the current user. If that attempt fails, this property attempts to get the domain name associated with the user name provided by the UserName property. If that attempt fails because the host computer is not joined to a domain, then the host computer name is returned.
```